homework02/rudolf/string.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+#include <iostream>
+#include <memory>
+#include <algorithm>
+
+std::size_t strlen(const char* chars);
```

Comment:
strlen is defined in `cstring`, this could potentially provoke a symbol clash, as the C runtime defines strlen in the global namespace, like you do